src/components/multi-select/multi-select.jsx

```diff
@@ -203,7 +201,7 @@ const MultiSelect = forwardRef(function MultiSelect(props, ref) {
 
   useEffect(() => {
     props.onChange({ target: selfRef.current });
-  }, [value.map(val => JSON.stringify(val)).join(',')]);
+  }, [value.join(',')]);
```

Comment:
Maybe revert.

Reply:
Honestly this works because users only ever perform a remove or add action. never a mix-match.